apps/mobile/src/components/FilterOffersScreen/index.tsx

```diff
@@ -3,23 +3,20 @@ import {type RootStackScreenProps} from '../../navigationTypes'
 import React from 'react'
 import FilterOffersContent from './FilterOffersContent'
 import FilterOffersScopeProvider from '../FilterOffersScopeProvider'
+import useSafeGoBack from '../../utils/useSafeGoBack'
 
 type Props = RootStackScreenProps<'FilterOffers'>
 
 function FilterOffersScreen({
   route: {
     params: {type},
   },
-  navigation,
 }: Props): JSX.Element {
+  const safeGoBack = useSafeGoBack()
   return (
     <Screen customHorizontalPadding={0} customVerticalPadding={32}>
       <FilterOffersScopeProvider type={type}>
-        <FilterOffersContent
-          navigateBack={() => {
-            navigation.goBack()
-          }}
-        />
+        <FilterOffersContent navigateBack={safeGoBack} />
```

Comment:
Great. It's better to use this. In the future we might open some screen directly by deepLink / notification and it might be problem calling goBack without checking if there is previous navigation state in the stack.